src/main/java/seedu/address/model/person/Item.java

```diff
@@ -0,0 +1,299 @@
+                .append(" Reminder: ")
+                .append(getReminder().toString())
+                .append(" Tags: ")
+                .append(getTags().toString());
+        getTags().forEach(builder::append);
+        return builder.toString();
+    }
+
+    /**
+     * Builder class for Item.
+     */
+    public static class ItemBuilder {
+
+        // Identity fields
+        private Task task;
+        private Event event;
+        private Reminder reminder;
+
+        // Data fields
+        private ItemDescription itemDescription;
+        private Set<Tag> tags = new HashSet<>();
+
+        public ItemBuilder() {}
+
+        public ItemBuilder setTask(Task task) {
+            requireNonNull(task);
+            this.task = task;
+            return this;
+        }
+
+        public ItemBuilder setEvent(Event event) {
+            requireNonNull(event);
+            this.event = event;
+            return this;
+        }
+
+        public ItemBuilder setReminder(Reminder reminder) {
+            requireNonNull(reminder);
+            this.reminder = reminder;
+            return this;
+        }
+
+        public ItemBuilder setItemDescription(ItemDescription descriptor) {
+            requireNonNull(descriptor);
+            this.itemDescription = descriptor;
+            return this;
+        }
+
+        //Consider using a defensive copy of tags, similar to EditCommand in AB3
+        public ItemBuilder setTags(Set<Tag> tags) {
+            requireNonNull(tags);
+            this.tags = tags;
+            return this;
+        }
+
+        /**Validates arguments of Item before initialising it
+         *
+         * @return A valid Item.
+         * @throws IllegalArgumentException If description not provided or task, event and reminder fields are null.
+         */
+        public Item build() throws IllegalArgumentException {
+            Item newItem = new Item(this.task, this.event, this.reminder, this.itemDescription, this.tags);
+
+            //Validation of parameters of object after object has been created.
+            //Validate after object has been created as per StackOverflow link
+            //https://stackoverflow.com/questions/38173274/builder-pattern-validation-effective-java
+            //However this seems to be contrary to the answer in the following link
+            //https://stackoverflow.com/questions/12930852/clearing-doubts-about-the-builder-pattern
+            //However it seems safer to follow the first as the object fields could be mutated after it has been copied
+            // from the builder to the object.
+            if (newItem.getItemDescription() == null) {
+                throw new IllegalArgumentException("Description must be provided!");
+            }
+            if (newItem.getTask() == null && newItem.getEvent() == null && newItem.getReminder() == null) {
+                throw new IllegalArgumentException("Task, Event & Reminder cannot all be empty!");
+            }
```

Comment:
Is there a reason why you do this check here instead of at the constructor level?